lib/libvarnishapi/vsm.c

```diff
@@ -682,7 +687,7 @@ vsm_refresh_set(struct vsm *vd, struct vsm_set *vs)
 		vs->vg = VTAILQ_FIRST(&vs->segs);
 		vs->fd = openat(vs->dfd, "_.index", O_RDONLY);
 		if (vs->fd < 0)
-			return (vs->retval | vs->flag_restarted);
+			return (vs->retval | restarted);
```

Comment:
You are now returning the restarted flag based solely on the directory stat changing.  Is reopening the file because of a file stat change also reason to return the restarted flag?

Reply:
What happens here is that needing to open the index implies a restart only if the directory had to be closed beforehand. If there was no directory to close it's the first start the VSM has seen, not a restart.

Reply:
Sure.  I wasn't talking about just opening the index, but rather reopening it because the file `stat` changed.  But if you've decided that isn't a restart, then maybe the description needs to be refined:  
> reopening the index means restarted

Reply:
There is no stat check for the index, the index is closed when the directory is not opened. And we close the directory when its stat changed, which should only happen when the cache is restarted. And now we've gone full circle :]

Reply:
After another look I see what you mean now. Indeed the commit message is misleading in this regard. I think I meant reopening the directory, because it's oddly missing from the list.